token-metadata/program/src/utils.rs

```diff
@@ -55,79 +54,103 @@ pub fn assert_data_valid(
     if data.seller_fee_basis_points > 10000 {
         return Err(MetadataError::InvalidBasisPoints.into());
     }
-    if data.creators.is_some() {
-        if let Some(creators) = &data.creators {
-            if creators.len() > MAX_CREATOR_LIMIT {
-                return Err(MetadataError::CreatorsTooLong.into());
-            }
 
-            if creators.is_empty() {
-                return Err(MetadataError::CreatorsMustBeAtleastOne.into());
+    if let Some(creators) = &data.creators {
+        if creators.len() > MAX_CREATOR_LIMIT {
+            return Err(MetadataError::CreatorsTooLong.into());
+        }
+
+        if creators.is_empty() {
+            return Err(MetadataError::CreatorsMustBeAtleastOne.into());
+        }
+
+        // Store caller-supplied creator's array into a hashmap for direct lookup.
+        let new_creators_map: HashMap<&Pubkey, &Creator> =
+            creators.iter().map(|c| (&c.address, c)).collect();
+
+        // Do not allow duplicate entries in the creator's array.
+        if new_creators_map.len() != creators.len() {
```

Comment:
Is it better to fail here or just handle it by removing the redundancy?

Reply:
Seems like better to fail because 1) That's how it currently works, and 2), if you sent a malformed array to me it indicates problems on the front end and I don't think we want to correct that automatically, maybe its hiding a problem?

Reply:
Yeah, you're right. Keep it as is.